syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

service Greeter {

  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloServerStream (HelloRequest) returns (stream HelloReply) {}

  rpc SayHelloClientStream (stream HelloRequest) returns (HelloReply) {}

  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}